misk/src/main/kotlin/misk/MiskServiceModule.kt

```diff
@@ -21,28 +21,35 @@ import misk.tokens.TokenGeneratorModule
  * with [MiskTestingServiceModule]. Only bindings that are not suitable for a unit testing
  * environment belong here.
  */
-class MiskRealServiceModule : KAbstractModule() {
+class MiskRealServiceModule(
+  private val serviceManagerConfig: ServiceManagerConfig,
+) : KAbstractModule() {
+  constructor(): this(ServiceManagerConfig())
```

Comment:
These secondary constructors should be good enough for backwards compatibility — anyone who wants to add this behaviour will need to explicitly opt in by providing a ServiceManagerConfig.

Reply:
I'd probably lean towards having the config param have a default config value than secondary constructor. A bit easier to reason about where default values are coming from.

Reply:
Sure I can change this to a single constructor with `@JvmOverloads`.

Reply:
This had an implication on the `projectHealth` check (see https://github.com/cashapp/misk/pull/3285/commits/fe2b1892c24c0faf857110b2af6f409afd94c283), but I think it's not a problem, since the dependency is still pulled in — seems like more of a misk samples specific linting error?